src/service/action-impl.js

```diff
@@ -152,11 +158,33 @@ export class ActionService {
           this.trigger(dev().assertElement(event.target), 'tap', event);
         }
       });
-    } else if (name == 'submit' || name == 'change') {
+    } else if (name == 'submit') {
       this.root_.addEventListener(name, event => {
         this.trigger(dev().assertElement(event.target), name, event);
       });
+    } else if (name == 'change') {
+      this.root_.addEventListener(name, event => {
+        const target = event.target;
+        const changeEvent = this.getChangeDetails_(event);
+        this.trigger(dev().assertElement(target), name, changeEvent);
+      });
+    }
+  }
+
+  getChangeDetails_(event) {
+    const details = {};
+    const target = event.target;
+    if (event.target.tagName.toLowerCase() === 'input') {
+      const inputType = target.getAttribute('type');
+      const fieldsToInclude = WHITELISTED_INPUT_DATA_[inputType];
+      // TODO(kmh287): Cast numeric values to numbers?
```

Comment:
@dvoytenko 

Just confirmed, `min` and `max` are strings. [Image](https://drive.google.com/a/ampproject.org/file/d/0B2blAKCEOPg-cm0yMEVCMmRVbjA/view?usp=sharing)